### ISSUE TITLE
after_create does not get the object created by to_create

### ISSUE THREAD

**Issue by TheWudu** [thoughtbot/factory_bot#1435]
### Description

When defining a factory including `to_create`, i would expect to get the response of this method/block also in a following `after_create`

E.g. 

```
FactoryBot.define do
  factory :user do
    to_create do |user|
      create_the_user(user)
    end 

    first_name                       { "John" }
    sequence(:last_name)             { |n| "Doe-#{n}" }

    trait :avatar do
      after(:create) do |object|
        object.avatar = build(:avatar, user_id: object.id)
        object.save!
      end
    end
end
```

Whatever create_the_user returns is NOT passed in as `object` then.

### Expected behavior

Pass return value of `to_create` into following `after_create` blocks.

### Actual behavior

The initial created object is passed into the `after_create` hooks

### System configuration
**factory_bot version**:  5.2.0
**rails version**: - 
**ruby version**: jruby-9.2.6.0

### Assumption:

I guess it is enough to assign the return of `evaluation.create(instance)` to `instance` again. But as i barely know the rest of the factory-bot code, this is just a guess.

File: lib/factory_bot/strategy/create.rb

**Comment by composerinteralia** [comment]
> I guess it is enough to assign the return of evaluation.create(instance) to instance again.

Yeah, I think that would work. I would definitely be open to exploring that change. Any interest in pushing up a PR with that change to lib/factory_bot/strategy/create.rb? If the existing tests pass, then I think we are good. We would probably also want one test to cover this specific use case to make sure we don't break anything in the future (I can help with that if you are unsure of how to write that test).

**Comment by TheWudu** [comment]
https://github.com/thoughtbot/factory_bot/pull/1437

**Comment by composerinteralia** [comment]
After seeing the PR, I don't think this is something we can support. It seems somewhat unusual to have a factory that builds an instance of one object, but then returns a different object. Perhaps it would make sense to build the user with factory_bot, and then to pass that user off to your `create_the_user` function outside of factory_bot. I don't know what `create_the_user` does, so it is hard to advise beyond that.

Thank you!
